### ISSUE TITLE
Stripe promotion codes stream bad request failure

### ISSUE THREAD

**Issue by jagannathsrs** [airbytehq/airbyte#11488]
## Environment
- **Airbyte version**:  0.33.11-alpha
- **OS Version / Instance**: AWS EKS
- **Deployment**: Kubernetes deploy env
- **Source Connector and version**: Stripe 0.1.30
- **Destination Connector and version**: Snowflake 0.4.24
- **Severity**: High
- **Step where error happened**:Sync Job

## Current Behavior
The whole sync fails if promotion codes stream is selected.

## Expected Behavior
It should sync the data or skip if there's no data

## Logs
[logs-27693-0.txt](https://github.com/airbytehq/airbyte/files/8366865/logs-27693-0.txt)
<details>
<summary>LOG</summary> 

</details>

**Comment by midavadim** [comment]
Hi,
I was trying to re-produce this error but was not able to do it:

All my syncs were successful:

1. I have tried exactly the same date 1582905600 (which is "start_date": "2020-02-28T16:00:00Z"), so url were exactly the same as in error from logs:

example from my sync:
    "https://api.stripe.com:443 \"GET /v1/promotion_codes?limit=100&created%5Bgte%5D=1582905600 HTTP/1.1\" 200 1956"}}

2. I have tried dates in the future (2025-02-28T16:00:00Z )
3. I have tried dates in the past (2000-02-28T16:00:00Z )

@jagannathsrs, will the problem happen if you change "Replication start date" to more recent?
